en/guide/routing.md

```diff
@@ -1,14 +1,81 @@
+The protocol defines two 8-bit fields that can (optionally) be specified in the message payload to indicate where the message should be sent/routed.
+
+- `target_system`: System that should execute the command
+- `target_component`: Component that should execute the command (requires `target_system`).
+
+The ids of the destination system and/or component should be set to the required target.
+If the ids are omitted or set to zero then the message is considered a *broadcast* (intended for all systems).
+
+MAVLink components are expected to process messages that have a matching system/component id and broadcast messages.
+They are expected to route/resend messages that are intended for other (or all) recipients to other active channels 
+(i.e. MAVLink systems may be connected across different transports, connected by a MAVLink system that routes the messages).
+Broadcast messages are forwarded to all channels that haven't seen the message. 
+Addressed messages are resent on a new channel *iff* the system has previously seen a message from the target on that channel 
+(messages are not resent if the addressee is not known or is on the original/incoming channel). 
+
+
+## Routing Detail
+
+Systems/components should process a message locally if any of these conditions hold:
+- It is a broadcast message (`target_system` field omitted or zero).
+- The `target_system` matches its system id and `target_component` is broadcast (`target_component` omitted or zero).
+- The `target_system` matches its system id and has the component's `target_component`
+- The `target_system` matches its system id and the component is unknown (i.e. this component has not seen any messages on any link that have the message's `target_system`/`target_component`).
+
+Systems should forward messages to another link if any of these conditions hold:
+- It is a broadcast message (`target_system` field omitted or zero).
+- The `target_system` does not match the system id *and* the system knows the link of the target system (i.e. it has previously seen a message from `target_system` on the link).
+- The `target_system` matches its system id and has a `target_component` field, and the system has seen a message from the `target_system`/`target_component` combination on the link.
+
+> **Note** Non-broadcast messages must only be sent (or forwarded) to known destinations (i.e. a system must previously have received a message from the target system/component). Systems should also clear stored information about a detected system/component if it detects a `SYSTEM_TIME` message with a decrease in `time_boot_ms`, as this indicates that the system has rebooted, and might have different routing information.
+
+Routed messages are *unchanged*; the source and destination ids are the values from the originating system.
+
+## Library Support
+
+### C Library (mavgen)
+
+The generated code for MAVLink v1 has no specific support for routing or working with `target_system` and `target_component`.
```

Comment:
This isn't MAVLink v1, this is just the C library. The above routing rules are pretty fixated on having this information readily available so what do you do in this case?